packages/transformer/src/BranchProvenanceInitializer.ts

```diff
@@ -24,27 +24,71 @@ export interface ProvenanceInitArgs {
    * insert Federation Guids in all lacking elements in the master database, which will prevent
    * needing to insert External Source Aspects for provenance tracking
    * @note requires a read/write master
+   * @note closes both the master and branch iModels to reset caches, so you must reopen them.
+   *       If you pass `"keep-reopened-db"`, this object's `master` and `branch` properties will
+   *       be set to new, open databases.
    */
-  createFedGuidsForMaster?: boolean;
+  createFedGuidsForMaster?: true | false | "keep-reopened-db";
 }
 
-interface ProvenanceInitResult {
+/**
+ * @alpha
+ */
+export interface ProvenanceInitResult {
   targetScopeElementId: Id64String;
+  masterExternalSourceId: Id64String;
+  masterRepositoryLinkId: Id64String;
 }
 
 /**
  * @alpha
  */
 export async function initializeBranchProvenance(args: ProvenanceInitArgs): Promise<ProvenanceInitResult> {
   if (args.createFedGuidsForMaster) {
-    // FIXME: elements in the cache could be wrong after this so need to purge cache somehow, maybe close the iModel
-    args.master.withSqliteStatement("UPDATE bis_Element SET FederationGuid=randomblob(16) WHERE FederationGuid IS NULL", (s) => {
-      assert(s.step() === DbResult.BE_SQLITE_DONE);
-    });
+    args.master.withSqliteStatement(`
+        UPDATE bis_Element
+        SET FederationGuid=randomblob(16)
+        WHERE FederationGuid IS NULL
+          AND Id NOT IN (0x1, 0xe, 0x10) -- ignore special elems
+      `,
+      (s) => assert(s.step() === DbResult.BE_SQLITE_DONE),
+    );
+    // NOTE: unfortunately attached database tables may not be used in an update query
+    // Couple other possibilities to test:
+    // - create a temporary table in the main tablespace and copy everything there, then update from that
```

Comment:
@khanaffan @ColinKerr 
any thoughts on improving this?

Reply:
after discussion, seems like a subquery makes the attached table approach work, but I can't seem to clear the cache to dump the writelock we make on the attached db, investigating.